parastell/magnet_coils.py

```diff
@@ -436,6 +409,87 @@ def create_magnet(self):
         shell = cq.Shell.makeShell(face_list)
         self.solid = cq.Solid.makeSolid(shell)
 
+    def in_toroidal_extent(self, lower_bound, upper_bound):
+        """Determines if the coil lies within a given toroidal angular extent,
+        based on filament coordinates.
+
+        Arguments:
+            lower_bound (float): lower bound of toroidal extent [rad].
+            upper_bound (float): upper bound of toroidal extent [rad].
+
+        Returns:
+            in_toroidal_extent (bool): flag to indicate whether coil lies
+                within toroidal bounds.
+        """
+        # Compute toroidal angle of each point in filament
+        toroidal_angles = np.arctan2(self._coords[:, 1], self._coords[:, 0])
+        # Ensure angles are positive
+        toroidal_angles = (toroidal_angles + 2 * np.pi) % (2 * np.pi)
+        # Compute bounds of toroidal extent of filament
+        min_tor_ang = np.min(toroidal_angles)
+        max_tor_ang = np.max(toroidal_angles)
+
+        # Determine if filament toroidal extent overlaps with that of model
+        if (min_tor_ang >= lower_bound or min_tor_ang <= upper_bound) or (
+            max_tor_ang >= lower_bound or max_tor_ang <= upper_bound
+        ):
+            in_toroidal_extent = True
+        else:
+            in_toroidal_extent = False
+
+        return in_toroidal_extent
+
+    def com_toroidal_angle(self):
+        """Computes the toroidal angle of the coil center of mass, based on
+        filament coordinates.
+
+        Returns:
+            (float): toroidal angle of coil center of mass [rad].
+        """
+        return np.arctan2(self.center_of_mass[1], self.center_of_mass[0])
+
+    def get_ob_mp_index(self):
+        """Finds the index of the outboard midplane coordinate on a coil
+        filament.
+
+        Returns:
+            outboard_index (int): index of the outboard midplane point.
+        """
+        # Compute radial distance of coordinates from z-axis
+        radii = np.linalg.norm(self.coords[:, :2], axis=1)
+        # Determine whether adjacent points cross the midplane (if so, they will
+        # have opposite signs)
+        midplane_flags = -np.sign(
+            self.coords[:, 2]
+            * np.append(self.coords[1:, 2], self.coords[1, 2])
+        )
```

Comment:
Maybe this makes things more readable?
```suggestion
        shifted_coords = np.append(self.coords[1:, :], self.coords[1, :])
        midplane_flags = -np.sign(self.coords[:, 2] * shifted_coords[:, 2])
```